js/src/dapps/registry/Events/events.js

```diff
@@ -162,33 +241,55 @@ export default class Events extends Component {
   }
 
   onReservedToggle = (e, isToggled) => {
-    const { pending, subscriptions, actions } = this.props;
+    const { pending, subscriptions, subscribe, unsubscribe } = this.props;
     if (!pending.Reserved) {
       if (isToggled && subscriptions.Reserved === null) {
-        actions.subscribe('Reserved');
+        subscribe('Reserved');
       } else if (!isToggled && subscriptions.Reserved !== null) {
-        actions.unsubscribe('Reserved');
+        unsubscribe('Reserved');
       }
     }
   };
   onDroppedToggle = (e, isToggled) => {
-    const { pending, subscriptions, actions } = this.props;
+    const { pending, subscriptions, subscribe, unsubscribe } = this.props;
     if (!pending.Dropped) {
       if (isToggled && subscriptions.Dropped === null) {
-        actions.subscribe('Dropped');
+        subscribe('Dropped');
       } else if (!isToggled && subscriptions.Dropped !== null) {
-        actions.unsubscribe('Dropped');
+        unsubscribe('Dropped');
       }
     }
   };
   onDataChangedToggle = (e, isToggled) => {
-    const { pending, subscriptions, actions } = this.props;
+    const { pending, subscriptions, subscribe, unsubscribe } = this.props;
     if (!pending.DataChanged) {
       if (isToggled && subscriptions.DataChanged === null) {
-        actions.subscribe('DataChanged');
+        subscribe('DataChanged');
       } else if (!isToggled && subscriptions.DataChanged !== null) {
-        actions.unsubscribe('DataChanged');
+        unsubscribe('DataChanged');
+      }
+    }
+  };
+
+  onReverseToggle = (e, isToggled) => {
+    const { pending, subscriptions, subscribe, unsubscribe } = this.props;
+
+    for (let e of ['ReverseProposed', 'ReverseConfirmed', 'ReverseRemoved']) {
+      if (pending[e]) {
```

Comment:
Return correct and non-squashed here, remainder of the code should follow this readable approach. (return/continue)